### PR TITLE
necessary changes related to watermarks and emit-on-window-close

### LINKED ISSUE #1117: Document APPEND ONLY clause for CREATE TABLE
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/11233

### Which part(s) of the docs might be affected or should be updated? And how?

Document APPEND ONLY clause for CREATE TABLE. If a table is APPEND ONLY, watermarks can be created.

### Reference

_No response_

### LINKED ISSUE #1136: Docs for watermark and `emit on window close`
### Related code PR

risingwavelabs/risingwave#9622 and many others.

### Which part(s) of the docs might be affected or should be updated? And how?

Now we have a [summary of Emit on window close](https://github.com/risingwavelabs/risingwave-docs/pull/1124) and watermark documents, some other docs can be updated accordingly and linked to them. 

Following docs need to be updated to reflect the watermark and `emit on window close`. Please mark them as experimental as well.

- [ ] [CREATE SOURCE](https://www.risingwave.dev/docs/current/sql-create-source/): needs to mention watermark definition
- [ ] [CREATE SINK](https://www.risingwave.dev/docs/upcoming/sql-create-sink/) needs to mention that `emit on window close` can help to build an append-only sink.
- [ ] [GROUP BY clause](https://www.risingwave.dev/docs/upcoming/query-syntax-group-by-clause/) needs to mention that `emit on window close` can help to make an append-only output and avoid some unnecessary computation
- [ ] Window functions (no docs yet) needs to mention that `emit on window close` can help to make an append-only output and avoid some unnecessary computation

### Reference

_No response_

### PR DESCRIPTION
**Pull request by hengm3467**
## Info

- **Description**

  Make necessary changes to related topics over watermarks and emit-on-window-close. The topic for watermarks was already published: https://www.risingwave.dev/docs/upcoming/emit-on-window-close/

The topic about window functions is in review: https://github.com/risingwavelabs/risingwave-docs/pull/1077

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves #1136

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Comment by CharlieSYH** [comment]
related: https://github.com/risingwavelabs/risingwave/pull/11454

File: docs/sql/commands/sql-create-sink.md [hunk 1]
```diff
@@ -65,30 +66,27 @@ rr.Stack(
...
 
- * [Kafka](/guides/create-sink-kafka.md) (Supports versions 3.1.0 or later)
- * JDBC-available databases
-   * [MySQL](/guides/sink-to-mysql.md) (Supports versions 5.7 and 8.0.x)
-   * [PostgreSQL](/guides/sink-to-postgres.md)
-   * [TiDB](/guides/sink-to-tidb.md)
- * [Apache Iceberg](/guides/sink-to-iceberg.md)
-
+* [Kafka](/guides/create-sink-kafka.md) (Supports versions 3.1.0 or later)
+* JDBC-available databases
```
Comment (stdrc) [inline docs/sql/commands/sql-create-sink.md]: Is this line still needed?

File: docs/sql/commands/sql-create-table.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ CREATE TABLE [ IF NOT EXISTS ] table_name (
     col_name data_type [ PRIMARY KEY ] [ AS generation_expression ],
     ...
     [ PRIMARY KEY (col_name, ... ) ]
+    [ watermark_clause ]
```
Comment (stdrc) [inline docs/sql/commands/sql-create-table.md:21]: For `CREATE TABLE`, user can only specify `watermark_clause` if `APPEND ONLY` is specified after schema definition, before `WITH`. For `CREATE SOURCE`, the `APPEND ONLY` is not needed. We need to update the syntax in doc. Here's an example: https://github.com/risingwavelabs/risingwave/blob/c426cbafbd783bcea53f29a700065fa0aca87f47/src/tests/state_cleaning_test/data/agg.toml#L7-L21

Comment (emile-00) [reply docs/sql/commands/sql-create-table.md:21]: Just for reference, the related doc issue for `APPEND ONLY` is here: https://github.com/risingwavelabs/risingwave-docs/issues/1117.

**Review by emile-00 — APPROVED** [review]

**Review by stdrc — APPROVED** [review]

**Pull request merged**